### PR TITLE
Overlay fix for default windows style and other light theme's.

### PR DESCRIPTION
**Pull request by MisterMakerNL**
These additions changes the Tree text color based on the stylesheet with a transparency state.
https://github.com/FreeCAD/FreeCAD/issues/12424
This is how it looks with transparency on
afbeelding
This is how it looks with transparency off.
afbeelding

Current build if you use black text in the tree, you would get this with transparency on: 
afbeelding
This is the reason we cannot do light theme's on dark background currently. This is also the reason I think overlay stylesheets are changing the background of normal widgets. So the text can stay white, in both states.

@kadet1090 Did all the difficult code stuff, I did the stylesheet.

**Still not working correctly as the mouse over doesn't trigger the color change.**

### CODE REVIEW

**Comment by pierreporte** [comment]
Does it fix #11340?

**Comment by MisterMakerNL** [comment]
@pierreporte I have no idea, it does clean it up a lot. You can test it by applying the added stylesheet https://github.com/FreeCAD/FreeCAD/pull/12573/commits/eb04a7bf36a85aff1e74a2faa1e1c7e823e2ce5c
@maxwxyz It would be wise to add some settings to the background settings panel, as these all are tied together. But that is a lot of work. 
Current option does work most of the time, as due to the issue with sketch lines colors being hard-coded it is not really advisable to use light backgrounds.

**Comment by kadet1090** [comment]
> @kadet1090 Is the text color setting exposed in the preferences or is there a general switch to Change overall light or dark within the same theme?

I did not touch any preference. I simply exposed "transparent" property for the overlay dock widget so it could be targeted via `Gui--OverlayTabWidget[transparent="true"]` or `Gui--OverlayTabWidget[transparent="false"]`. The color itself is controlled via the stylesheets in that case.

**Comment by maxwxyz** [comment]
@Rexbas you've currently made some fixed to mouse over/click through with the overlay panels active, do you know a fix?

**Comment by MisterMakerNL** [comment]
So this fix works pretty great, and should fix the windows overlay style issues.
The only issue here that stops this for being merged is that it doesn't work with the show on mouse function aka mouse over.
And so it breaks this function for light theme's, and I have no clue on how to fix it.

**Comment by Rexbas** [comment]
I would like to take a look, but what exactly is the problem, can someone show a video and/or reproduction steps? If I understand correctly this problem happens on Windows machines. Does it also happen on Linux? I prefer not having to setup development on Windows.

**Comment by MisterMakerNL** [comment]
Here you see the issue. With this PR the black text turns white when overlay transparency is active. 
This is done by the overlay stylesheet with [transparent="true"] tag.
But as you can also see in the gif, this doesn't work when hoover over is active, the text color stays dark.
You can also see how much better it looks with white text on the default background.

These are windows default settings, so it is a big issue.

FreeCAD_xkozxvj2du

**Comment by Rexbas** [comment]
@MisterMakerNL, I don´t understand style sheets. When I change something in src/Gui/Stylesheets nothing happens even if I remove files. How do change something and see the result?

**Comment by MisterMakerNL** [comment]
You can use the added [src/Gui/Stylesheets/overlay/Default_Light.qss](https://github.com/FreeCAD/FreeCAD/pull/12573/files#diff-a9adccf0b767eb0087a8f46c70f9c83a5742efdf1a9ae15507907ea207db0ea5)
I forgot too add it too the cmakelist....
But if you enable that in the overlay stylesheet, it should work in this pr.
image

**Comment by Rexbas** [comment]
@MisterMakerNL, try with this c06bb1bcce0edd0c5e729228ca49be4d714e9367 from my branch (https://github.com/Rexbas/FreeCAD/tree/overlay-transparent-property)

**Comment by Rexbas** [comment]
Also if it works then we might be able to drop commit 0e5e7559b4299f3b6fe535d1ad81d3bc4ae3a036

**Comment by MisterMakerNL** [comment]
Compiled fine but sadly it didn't work. :(

**Comment by MisterMakerNL** [comment]
Seems to work now! 
FreeCAD_oJdsCC2O4S

**Comment by MisterMakerNL** [comment]
I added more overlay sheets, but I am having huge lagspikes not sure if this is caused by this PR.

**Comment by Rexbas** [comment]
I have no lag, works fine. Could you revert 0e5e7559b4299f3b6fe535d1ad81d3bc4ae3a036 because setting the transparent property is no longer needed in those places. I moved it to a different place in 314f778ba16f7acb482cd097ed997470acaab311.

**Comment by MisterMakerNL** [comment]
Did it, tested, works fine!

File: src/Gui/OverlayWidgets.cpp [hunk 1]
```diff
@@ -1152,7 +1148,6 @@ void OverlayTabWidget::setTransparent(bool enable)
         return;
     if(hGrp) {
         Base::StateLocker lock(_saving);
-        hGrp->SetBool("Transparent", enable);
```
Comment (Rexbas) [inline src/Gui/OverlayWidgets.cpp:1150]: Small mistake here. Remove the line containing `setProperty("transparent", enable);` which is a couple of lines higher and add this line back.

Comment (MisterMakerNL) [reply src/Gui/OverlayWidgets.cpp]: done

**Comment by MisterMakerNL** [comment]
The reason there are 4 stylesheets is that it also sets the title bar icons, and we have light and dark icons. Might be a good idea too try to move these icons to the normal stylesheet, then we only have 2 overlay stylesheets, one for a light background and one for a dark background.

I think in the future it would be feasible to set the overlay stylesheet automatically based on the color of the background. Although difficult here would be that we have a gradient and if you add a light and dark color then it will break the code.

**Pull request merged**